test/apps/iodemo/io_demo.cc

```diff
@@ -953,6 +948,18 @@ class P2pDemoCommon : public UcxContext {
         validate(conn, msg, iomsg_size);
     }
 
+    inline void handle_io_msg(const iomsg_t *msg, UcxConnection *conn) {
+#ifndef NDEBUG
+        if (opts().use_am && (msg->op == IO_READ_COMP)) {
+            assert(conn->id() == msg->conn_id);
+        } else {
+            assert((conn->remote_id() == 0) ||
+                   (conn->remote_id() == msg->conn_id));
```

Comment:
as discussed offline this assert should be different for TAG and AM cases

Reply:
it seems that the current state works fine

Reply:
it works but remote_id should be initialized for TAG

Reply:
fixed